skymp5-server/cpp/lib_espm/GroupUtils.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include "espm.h"
+
+namespace espm {
+
+template <class TBrowser>
+const GroupHeader* GetExteriorWorldGroup(const TBrowser& browser,
+                                         const RecordHeader* rec)
+{
+  for (auto gr : browser.GetParentGroupsEnsured(rec)) {
+    if (gr->GetGroupType() == GroupType::WORLD_CHILDREN) {
+      return gr;
+    }
+  }
+  return nullptr;
+}
+
+template <class TBrowser>
+const GroupHeader* GetCellGroup(const TBrowser& browser,
+                                const RecordHeader* rec)
+{
+  for (auto gr : browser.GetParentGroupsEnsured(rec)) {
+    auto grType = gr->GetGroupType();
+    if (grType != GroupType::CELL_CHILDREN &&
+        grType != GroupType::CELL_PERSISTENT_CHILDREN &&
+        grType != GroupType::CELL_TEMPORARY_CHILDREN &&
+        grType != GroupType::CELL_VISIBLE_DISTANT_CHILDREN) {
+      continue;
+    }
+    return gr;
+  }
+  return nullptr;
+}
+
+template <class TBrowser>
+uint32_t GetWorldOrCell(const TBrowser& browser, const RecordHeader* rec)
+{
+  const auto world = GetExteriorWorldGroup(browser, rec);
+  const auto cell = GetCellGroup(browser, rec);
+
+  uint32_t worldOrCell;
+
+  if (!world || !world->GetParentWRLD(worldOrCell)) {
+    worldOrCell = 0;
+  }
+
+  if (!worldOrCell) {
+    if (!cell->GetParentCELL(worldOrCell)) {
+      return 0;
+    }
+  }
+
```

Comment:
Originally authored by me, but I think it's time to refactor this:
```c++
if (world && world->GetParentWRLD(worldOrCell)) {
  return worldOrCell;
}
if (cell && cell->GetParentCELL(worldOrCell)) {
  return worldOrCell;
}
return 0;
```
But please make sure it's covered by tests.

Reply:
I don't think that such long `if`s (`if (const auto ... = ...; ... ... ...)`) are good, but overall seems better

Reply:
Feel free to move `const auto` from condition

Reply:
> But please make sure it's covered by tests.

Yep, it is. I've replaced this function with just `return 0` and tests failed.

Reply:
I think it's better make everything a bit more consistent with both `if`s having same structure. Pushed that version to PR

Reply:
@nic11 does it work after refactor?

Reply:
Yes